src/actions/waitforexternaleventaction.ts

```diff
@@ -1,12 +1,14 @@
-import { ActionType, IAction, Utils } from "../classes";
+import { ActionType, ExternalEventType, IAction, Utils } from "../classes";
 
 /** @hidden */
 export class WaitForExternalEventAction implements IAction {
     public readonly actionType: ActionType = ActionType.WaitForExternalEvent;
 
     constructor(
         public readonly externalEventName: string,
+        public readonly reason: ExternalEventType,
     ) {
         Utils.throwIfEmpty(externalEventName, "externalEventName");
+        Utils.throwIfEmpty(reason, "reason");
```

Comment:
`reason` is a completely internal representation of how we store events that is new in `2.0`. We will need to default it (I will to "ExterenalEvent") so that this new npm package won't break with `1.x` versions of the Durable Functions extension.